### PR TITLE
Fixes coverity issues, suppress Serialization issues

### PR DESCRIPTION
**Pull request by Siedlerchr**
- [X ] Changes in pull request outlined? (What, why, ...)

Used Lambdas for ActionListener
Fixed resource Leak in DB Exporter Methods

### CODE REVIEW

File: src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java [hunk 1]
```diff
@@ -15,23 +15,23 @@
...
+import net.sf.jabref.groups.UndoableChangeAssignment;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.util.strings.QuotedStringTokenizer;
+import net.sf.jabref.logic.util.strings.StringUtil;
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.entry.BibEntry;
+
 /**
- * Select explicit bibtex entries. It is also known as static group.
+ * Select explicit bibtex groupEntries. It is also known as static group.
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java]: Why did you rename entries to group entries? It is relatively clear from the context that these entries belong to the explicit group.

File: src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java [hunk 2]
```diff
@@ -217,7 +217,7 @@ public String getDescription() {
     }
 
     public static String getDescriptionForPreview() {
-        return Localization.lang("This group contains entries based on manual assignment. "
+        return Localization.lang("This group contains groupEntries based on manual assignment. "
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java]: you definitely don't want this change.

File: src/main/java/net/sf/jabref/gui/GroupAddRemoveDialog.java [hunk 3]
```diff
@@ -51,16 +57,16 @@ public GroupAddRemoveDialog(BasePanel panel, boolean add, boolean move) {
     @Override
     public void action() throws Throwable {
         GroupTreeNode groups = panel.getBibDatabaseContext().getMetaData().getGroups();
-        if (groups == null) {
+        if (Objects.isNull(groups)) {
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/gui/GroupAddRemoveDialog.java]: Objects.isNull exists more or less just to be used as a predicate (e.g. filter(Objects::isNull). The previous version with == null is more readable in general.

File: src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java]: I'd guess it is because some methods use `entries` as variables, see e.g. line 82.

@Siedlerchr I've tried to avoid touching the groups package since there's a big PR in the waiting to be merged after 3.3 is released, #847, just to avoid conflicts there.

File: src/main/java/net/sf/jabref/gui/util/comparator/IconComparator.java [hunk 4]
```diff
@@ -17,13 +17,14 @@
...
 
+import java.io.Serializable;
 import java.util.Comparator;
 import java.util.List;
 
 /**
  * Comparator that handles icon columns.
  */
-public class IconComparator implements Comparator<BibEntry> {
+public class IconComparator implements Comparator<BibEntry>, Serializable {
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/gui/util/comparator/IconComparator.java]: I think the Serializable is not necessary (same also below)

File: src/main/java/net/sf/jabref/gui/dbproperties/SaveActionsPanel.java [hunk 5]
```diff
@@ -138,19 +134,22 @@ public void storeSettings(MetaData metaData) {
     }
 
     public boolean hasChanged() {
-        List<FieldFormatterCleanup> newActions = ((SaveActionsListModel) actionsList.getModel()).getAllActions();
+        List<FieldFormatterCleanup> newActions = ((SaveActionsListModel<FieldFormatterCleanup>) actionsList.getModel())
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/dbproperties/SaveActionsPanel.java]: Do you need to specify `FieldFormatterCleanup` on the right hand side or is it enough to write `<>`? I have no idea...

**Comment by tobiasdiez** [comment]
In general, this looks good to me. But I would welcome if you wait with merging this in until #847 is accepted (because otherwise rebasing this other PR will be a hell)

File: src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java (hunk 2 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java]: That is probably what is causing the failing test as well.

File: src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java (hunk 1 above)
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java]: I renamed it, because nearly every method uses entries and therefore would be hiding/overriding this.

File: src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java (hunk 2 above)
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java]: Oops. I oversaw this from the Rename. Will correct it.

File: src/main/java/net/sf/jabref/gui/GroupAddRemoveDialog.java (hunk 3 above)
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/GroupAddRemoveDialog.java]: Okay. Noted. Will change it back.

File: src/main/java/net/sf/jabref/gui/util/comparator/IconComparator.java (hunk 4 above)
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/util/comparator/IconComparator.java]: I meant not to change this. Seems like the files were not correctly overwritten when I deleted the old branch from the previous PR.

File: src/main/java/net/sf/jabref/gui/dbproperties/SaveActionsPanel.java (hunk 5 above)
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/dbproperties/SaveActionsPanel.java]: In this case it's necessary, it can't be derived automatically.
Previously all references to the type where non-generic

**Comment by Siedlerchr** [comment]
@tobiasdiez  No problem.  What about including the few changes to the group related classes here in your PR?

File: src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java (hunk 1 above)
Comment (simonharrer) [inline src/main/java/net/sf/jabref/groups/structure/ExplicitGroup.java]: Javadoc should be human readable, and not contain internal field names. Hence, I vote for entries instead of the internal groupEntries.

**Comment by stefan-kolb** [comment]
We should be able to merge this now.

**Comment by oscargus** [comment]
Yes, but quite a few of the files has been moved since the original PR... I would almost suggest redoing compared to rebasing/merging... Good luck @Siedlerchr! :+1:

**Comment by Siedlerchr** [comment]
Yeah, I will look at this in the next days. I think many of these things already have been taken into account.

**Comment by koppor** [comment]
Bump :innocent:

**Comment by Siedlerchr** [comment]
@koppor  Re-added the changes. Some issue were already fixed.
